xarray/core/computation.py

```diff
@@ -991,43 +941,49 @@ def earth_mover_distance(first_samples,
 
     if kwargs is None:
         kwargs = {}
-
     signature = _UFuncSignature(input_core_dims, output_core_dims)
 
     if exclude_dims and not exclude_dims <= signature.all_core_dims:
         raise ValueError(
             "each dimension in `exclude_dims` must also be a "
             "core dimension in the function signature"
         )
+    # handle dask_gufunc_kwargs
+    if (dask == "parallelized") or vectorize:
+        if dask_gufunc_kwargs is None:
+            dask_gufunc_kwargs = {}
+        # needed for correctly distributing keywords to vectorize
+        dask_gufunc_kwargs.setdefault("output_dtypes", output_dtypes)
+        dask_gufunc_kwargs.setdefault("vectorize", vectorize)
+        # todo: remove warnings after deprecation cycle
+        if meta is not None:
+            warnings.warn(
+                "``meta`` should be given in the ``dask_gufunc_kwargs`` parameter."
+                " It will be removed as direct parameter in a future version."
+            )
+            dask_gufunc_kwargs.setdefault("meta", meta)
+        if output_sizes is not None:
+            warnings.warn(
+                "``output_sizes`` should be given in the ``dask_gufunc_kwargs`` "
+                "parameter. It will be removed as direct parameter in a future "
+                "version."
+            )
+            dask_gufunc_kwargs.setdefault("output_sizes", output_sizes)
```

Comment:
If we use `to_gufunc_string()` in generating the signature for passing into dask, then we won't be able to entirely defer handling of `output_sizes`. So I think we should probably keep it as our own argument?

Reply:
Sorry, I'm not sure I understand.

Reply:
I think Stephan refers to the problem you ran into with `stack_negative`.

Reply:
Would it be worth to combine `output_core_dims` and `output_sizes` into one?

Like

```python
output_core_dims=[[("sign", 2)]]
```
or

```python
output_core_dims=[{"sign": 2}]
```

Reply:
I copy my comment over for completeness:

> `output_core_dims=[{"abc": 2}]` looks very elegant. But it is passed to `_UFuncSignature(input_core_dims, output_core_dims)` so you would probably need to take it apart again in the code (or adapt `_UFuncSignature` to handle this case). Not sure how involved this is. Could potentially be left for a future PR?